internal/provider/resource_image_repo.go

```diff
@@ -173,6 +179,7 @@ func (r *imageRepoResource) Read(ctx context.Context, req resource.ReadRequest,
 	state.ID = types.StringValue(repo.Id)
 	state.ParentID = types.StringValue(uidp.Parent(repo.Id))
 	state.Name = types.StringValue(repo.Name)
+	state.Readme = types.StringValue(repo.Readme)
```

Comment:
Since Readme is optional, this may cause an inconsistent state when reading in a repo that doesn't have a Readme. It comes down to the fact that terraform types can have a value, or be `Null`, which is treated differently than `""`. 

I think this needs to only update `state.Readme` if we get back a non-empty string. Most of the resources that have a `Description` do this, you can check those for reference. It'd be something like

```
// Only update the state readme if it started as non-null or we receive a description.
if !(state.Readme.IsNull() && repo.Readme == "") {
	state.Readme = types.StringValue(repo.Readme)
}
```